### PR TITLE
[misc][installation] build from source without compilation

### PR DESCRIPTION
**Pull request by youkaichao**
make life easier for people who only want to change python files.

### CODE REVIEW

**Comment by dtrifiro** [comment]
Wouldn't using `sccache` achieve a similar goal?

**Comment by youkaichao** [comment]
`sccache` requires exactly the same compilation environment, while our wheel can be used regardless of the cuda toolkit.

e.g. a user who does not have `nvcc` installed, can use this method to develop vllm's python code.

File: docs/source/getting_started/installation.rst [hunk 1]
```diff
@@ -58,13 +58,41 @@ You can install vLLM using pip:
...
+---------------------------------------
+
+If you want to develop vLLM, and you only need to change the Python code, you can build vLLM without compilation.
+
+The first step is to follow the previous instructions to install the latest vLLM wheel:
+
+.. code-block:: console
+
+    $ export VLLM_VERSION=0.6.1.post1
+    $ pip install https://vllm-wheels.s3.us-west-2.amazonaws.com/nightly/vllm-${VLLM_VERSION}-cp38-abi3-manylinux1_x86_64.whl
```
Comment (tlrmchlsmth) [inline docs/source/getting_started/installation.rst:71]: I think we could find a way to make this a little easier -- either by wrapping this in a script that's version invariant, or maybe we could always upload the latest wheel to `https://vllm-wheels.s3.us-west-2.amazonaws.com/nightly/vllm-latest-cp38-abi3-manylinux1_x86_64.whl`

**Review by tlrmchlsmth — COMMENTED** [review]
I like this approach

File: docs/source/getting_started/installation.rst (hunk 1 above)
Comment (youkaichao) [reply docs/source/getting_started/installation.rst:71]: cc @simon-mo @khluu if this is possible

Comment (khluu) [reply docs/source/getting_started/installation.rst:71]: maybe push to `nightly` during the scheduled build at 9PM daily? or we can push to `latest/vllm-{VLLM_VERSION}...` for every commit

Comment (tlrmchlsmth) [reply docs/source/getting_started/installation.rst:71]: Ideally we shouldn't have to update this when we put a new version out, and ideally users could have one command that always. works.

Isn't `export VLLM_VERSION=0.6.1.post1` already out of date, btw?

**Review by tlrmchlsmth — APPROVED** [review]
I tried it out, works well! Also I realized that the installation guide already tells users to `export VLLM_VERSION=0.6.1.post1`, in a bunch of places so I'm OK with landing as-is

File: docs/source/getting_started/installation.rst (hunk 1 above)
Comment (simon-mo) [reply docs/source/getting_started/installation.rst:71]: lastest is always pushed. but the version string will always be there. a thing I have known worked in the pass is to set dev versions something like v1.0.0.dev0

Comment (youkaichao) [reply docs/source/getting_started/installation.rst:71]: can we omit the version in the wheel name? i.e. a unified link that always points to the latest wheel. an url re-direction should work I think.

**Pull request merged**